### PR TITLE
[wip] libomp updates

### PR DESCRIPTION
**Pull request by mgorny**
@voyageur @zzlei

### CODE REVIEW

**Comment by zzlei** [comment]
Possibly relevant: https://github.com/gentoo/gentoo/pull/2118

**Comment by voyageur** [comment]
Looks good to me (sorry for the delay)

**Comment by mgorny** [comment]
@voyageur, thanks. I plan to try to get tests working but I guess I'll work on that later.

@zzlei, if you could fine some time for it, could you prepare 3.9.0 libomp bump? The old alias removal patch no longer applies, so you'd have to either forward-port it or backport my upstream fix giving the CMake option.

**Comment by mgorny** [comment]
Pushed the live ebuild and updated the PR with my later WIP for tests (still incomplete though).

**Comment by zzlei** [comment]
Please give me a few days.

File: sys-libs/libomp/libomp-9999.ebuild [hunk 1]
```diff
@@ -17,16 +18,26 @@ EGIT_REPO_URI="http://llvm.org/git/openmp.git
...
+pkg_setup() {
+	use test && python-any-r1_pkg_setup
+}
 
 multilib_src_configure() {
 	local libdir="$(get_libdir)"
 	local mycmakeargs=(
 		-DLIBOMP_LIBDIR_SUFFIX="${libdir#lib}"
 		# do not install libgomp.so & libiomp5.so aliases
 		-DLIBOMP_INSTALL_ALIASES=OFF
```
Comment (zzlei) [inline sys-libs/libomp/libomp-9999.ebuild]: I guess libgomp is from gcc and libiomp5 from icc. Why do we install aliases for them in the first place? Isn't libomp supposed to used by clang exclusively?

File: sys-libs/libomp/libomp-9999.ebuild [hunk 2]
```diff
@@ -17,16 +18,26 @@ EGIT_REPO_URI="http://llvm.org/git/openmp.git
...
+	test? ( ${PYTHON_DEPS} )"
+
+pkg_setup() {
+	use test && python-any-r1_pkg_setup
+}
 
 multilib_src_configure() {
 	local libdir="$(get_libdir)"
 	local mycmakeargs=(
 		-DLIBOMP_LIBDIR_SUFFIX="${libdir#lib}"
```
Comment (zzlei) [inline sys-libs/libomp/libomp-9999.ebuild]: Why not `LLVM_LIBDIR_SUFFIX`? I recall you intend to use this consistently throughout all llvm components?

File: sys-libs/libomp/libomp-9999.ebuild [hunk 3]
```diff
@@ -17,16 +18,26 @@ EGIT_REPO_URI="http://llvm.org/git/openmp.git
 LICENSE="UoI-NCSA"
 SLOT="0"
 KEYWORDS=""
-IUSE=""
+IUSE="test"
 
-DEPEND="dev-lang/perl"
+DEPEND="dev-lang/perl
```
Comment (zzlei) [inline sys-libs/libomp/libomp-9999.ebuild]: Why isn't this needed in the non-live ebuild?

File: sys-libs/libomp/libomp-9999.ebuild (hunk 1 above)
Comment (mgorny) [inline sys-libs/libomp/libomp-9999.ebuild]: It's upstream invention. In <9999, we patched it out. Now we've upstream switch for it.

File: sys-libs/libomp/libomp-9999.ebuild (hunk 2 above)
Comment (mgorny) [inline sys-libs/libomp/libomp-9999.ebuild]: openmp ain't using LLVM cmake files at all.

File: sys-libs/libomp/libomp-9999.ebuild (hunk 3 above)
Comment (mgorny) [inline sys-libs/libomp/libomp-9999.ebuild]: It is needed. I simply didn't get to other ebuilds yet.

**Comment by mgorny** [comment]
And bam! I've fixed this and committed, and now we've got working tests in libomp-9999!

**Pull request closed without merging**